history.go

```diff
@@ -0,0 +1,75 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+type History struct {
+	tree          *MerkleTree
+	currentSTR    *SignedTreeRoot
+	epochInterval int64
+}
```

Comment:
I don't really disagree with you, though. It might make sense to rename ``currentSTR`` to ``str`` if what the ```History``` struct represents is a link or snapshot (or block if we want to go there) in the server's history. And to make this more clear, I might even be inclined to rename ``History`` to something more accurate. What do you think?